great_expectations/self_check/util.py

```diff
@@ -1029,24 +1028,41 @@ def build_sa_validator_with_data(  # noqa: C901 - 39
     else:
         sql_insert_method = None
 
-    _debug("Calling df.to_sql")
-    _start = time.time()
-    add_dataframe_to_db(
-        df=df,
-        name=table_name,
-        con=engine,
-        index=False,
-        dtype=sql_dtypes,
-        if_exists="replace",
-        method=sql_insert_method,
-    )
-    _end = time.time()
-    _debug(
-        f"Took {_end - _start} seconds to df.to_sql for {sa_engine_name} {extra_debug_info}"
+    execution_engine = SqlAlchemyExecutionEngine(caching=caching, engine=engine)
+    batch_data = SqlAlchemyBatchData(
+        execution_engine=execution_engine, table_name=table_name
```

Comment:
Passing execution engine and not engine to batch_data.

Reply:
YES